src/search/model/SearchModel.js

```diff
@@ -85,14 +106,32 @@ export class SearchModel {
 		if (result.restriction === restriction) { // both are the same instance
 			return false
 		}
-		return !isSameTypeRef(restriction.type, result.restriction.type)
-			|| restriction.start !== result.restriction.start
-			|| restriction.end !== result.restriction.end
-			|| restriction.field !== result.restriction.field
-			|| restriction.listId !== result.restriction.listId
+
+		return !compareSearchRestriction(restriction, result.restriction)
 	}
 }
 
+function areSearchQueriesEqual(a: SearchQuery, b: SearchQuery) {
+	return a.query === b.query
+		&& compareSearchRestriction(a.restriction, b.restriction)
+		&& a.minSuggestionCount === b.minSuggestionCount
+		&& a.maxResults === b.maxResults
+}
+
+function compareSearchRestriction(a: SearchRestriction, b: SearchRestriction): boolean {
```

Comment:
`areSearchQueriesEqual`, `compareSearchRestriction`, `arrayEquals` do the same thing for different types but named differently?

Reply:
fair point, i forget to rename `compareSearchRestriction` anyway. I'll make them all be named `*Equals`